packages/client/src/widgets/Game/ui/Game/GameOverModal/GameOverModal.tsx

```diff
@@ -0,0 +1,48 @@
+import { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { Modal, Button } from 'antd';
+import classes from './GameOverModal.module.scss';
+import { EAppRoutes } from '@/shared/types';
+import { IGameOverModalProps } from './GameOverModal.interfaces';
+import { TEXTS } from './GameOverModal.constants';
+
+export const GameOverModal: React.FC<IGameOverModalProps> = ({ onClose, score }) => {
+  const [visible, setVisible] = useState(true);
+  const navigate = useNavigate();
+  const handleClose = (): void => {
+    setVisible(false);
+    if (onClose) {
+      onClose();
```

Comment:
Можно убрать if и сделать так
```suggestion
      onClose?.();
```